roofit/roofit/src/RooLagrangianMorphFunc.cxx

```diff
@@ -0,0 +1,3384 @@
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// config setter for name of the the observable to be morphed
+void RooLagrangianMorphFunc::Config::setObservableName(const char* obsname){
+  this->_obsName = obsname;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// config setter for name of the the observable to be morphed
+void RooLagrangianMorphFunc::Config::allowNegativeYields(bool allowNegativeYields){
+  this->_allowNegativeYields = allowNegativeYields;
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// config setter for diagrams
+template <class T> 
+void RooLagrangianMorphFunc::Config::setDiagrams(const std::vector<std::vector<T> >& diagrams)
+{
+  for(size_t j=0; j<diagrams.size(); ++j){
+    std::vector<RooArgList*> vertices;
+    for(size_t i=0; i<diagrams[j].size(); i++){
+      vertices.push_back(new RooArgList());
+      vertices[i]->add(diagrams[j][i]);
+    }
+    this->_configDiagrams.push_back(vertices);
+ }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// config setter for vertices
+template <class T> 
+ void RooLagrangianMorphFunc::Config::setVertices(const std::vector<T>& vertices)
+{
+  std::vector<std::vector<T> > diagrams;
+  diagrams.push_back(vertices);
+  this->setDiagrams(diagrams);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// append the parameter map with a parmeter set
+
+void RooLagrangianMorphFunc::Config::append(RooLagrangianMorphFunc::ParamMap& map, const char* str, RooLagrangianMorphFunc::ParamSet& set)
+{
+  map[str]=set;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// set values to paramerter set (-?-)
+
+void RooLagrangianMorphFunc::Config::append(RooLagrangianMorphFunc::ParamSet& set, const char* str, double val)
+{
+  set[str]=val;
+}
+
+template void RooLagrangianMorphFunc::Config::setVertices<RooArgSet>(const std::vector<RooArgSet>&);
+template void RooLagrangianMorphFunc::Config::setDiagrams<RooArgSet>(const std::vector<std::vector<RooArgSet>>&);
+template void RooLagrangianMorphFunc::Config::setVertices<RooArgList>(const std::vector<RooArgList>&);
+template void RooLagrangianMorphFunc::Config::setDiagrams<RooArgList>(const std::vector<std::vector<RooArgList>>&);
+
+////////////////////////////////////////////////////////////////////////////////
+/// config setter for vertices
+void RooLagrangianMorphFunc::Config::disableInterference(const std::vector<const char*>& nonInterfering){
+  // disable interference between the listed operators
+  std::stringstream name;
+  name << "noInteference";
+  for(auto c:nonInterfering){
+    name << c;
+  }
+  RooArgList* p = new RooArgList(name.str().c_str());
+  this->_nonInterfering.push_back(new RooArgList());
+  for(auto c:nonInterfering){
+    p->add(*(new RooStringVar(c,c,c)));
+  }
```

Comment:
There's something missing here, pretty sure @rahulgrit forgot to move a piece of code. What is `p` here is actually supposed to be a member of the class I think (proper name something like `interferenceBlacklist`)

Reply:
Sorry for being late to the party, but why is this marked as "resolved" now, with the function just being commented completely?

Reply:
Hi @cburgard , I would like to fix this for good once the interface of the config class is finalised. I have commented it out for now.

Reply:
Hi @cburgard , this option was commented out for now as i thought it would be better to resolve this for good with the PR that would include updates to to the config interface.